middle_end/flambda2/terms/flambda_primitive.ml

```diff
@@ -1216,9 +1225,11 @@ let compare_binary_primitive p1 p2 =
     if c <> 0 then c else Stdlib.compare comp_behaviour1 comp_behaviour2
   | Float_arith op1, Float_arith op2 -> Stdlib.compare op1 op2
   | Float_comp comp1, Float_comp comp2 -> Stdlib.compare comp1 comp2
+  | Get_method { is_self = is_self1 }, Get_method { is_self = is_self2 } ->
+    Stdlib.compare is_self1 is_self2
```

Comment:
maybe add a `bool` annotation here to ensure that if the type ever changes, this code will be looked at ?